### ISSUE TITLE
Triggering submit with enter?

### ISSUE THREAD

**Issue by jaredatch** [select2/select2#1456]
I'm running s2 on an input field that grabs data via ajax.

For the life of me I can't figure out how to have the form submit on enter keypress, when a input value is picked. Using various js method didn't seem to work either, I assume because of how s2 handles the input.

What am I missing?

**Comment by d4mation-** [comment]
I ended up commenting out the code @jayschab mentioned and then used this to trigger the Form's Submission within my modal:
```javascript
$(document).ready(function() {
       $("#modal_id .dropdown").select2({
              width: "100%",
              openOnEnter: false, // killEvent removed from select2.js
       });
       $('#modal_id .select2-focusser').bind('keypress', modalEnter);
       function modalEnter(e){
            if(e.keyCode === 13){
                    $('#modal_id .save').click();
            }
       }
});
```

It is a bit quick and dirty, but it works. Hope to see a better solution like the filterEvent @ivaynberg mentioned integrated soon.

**Comment by martynchamberlin** [comment]
For me, just having openOnEnter set to false correctly submits the form. We're using the Select2's [Ajax method](http://select2.github.io/select2/#ajax) though, so our dropdown isn't pre-populated.

**Comment by reichertm** [comment]
In 4.0.0-beta.3 the following did the trick:
```
  $('.select2-selection').on('keypress', function (e) {
    if (e.keyCode === 13) {
      $(this).closest('form').submit();
    }
  });
```
Plus I removed the behavior of opening the select box on enter modifying the source of the `keypress` handler in `Select2.prototype._registerEvents`.

**Comment by kevin-brown** [comment]
Dropping this from the 4.0 milestone as workaround exist and this is relatively low priority.

**Comment by notacouch** [comment]
For Google Chrome (and it alone, apparently) hitting enter whilst on a select submits the form. I don't know if that has always been the case or for how long. Just figured I'd throw this in here if it wasn't already known.

**Comment by kevin-brown** [comment]
@notacouch I personally can't reproduce that on Chrome 40 on Ubuntu 14.04, though the possibility of different behavior on different operating systems is not out of the question.

**Comment by notacouch** [comment]
@kevin-brown Oh. That's interesting. I'm on OS X 10.10.2, colleague on OS X 10.9.5 also gets the same behavior. On Windows 7, Chrome opens up the dropdown in the same position. Real interesting.

Safari behaves similarly to Chrome for me, Firefox does not, nothing happens when hitting enter.

Not sure if there's a protocol for handling all these kinds of differences, in the least it's nice to know.

**Comment by digitalRain** [comment]
If you need this functionality, but don't want to change the select2.js, you can use the code provided by @reichertm with a slight change:
```javascript
    $('.select2-selection').on('keyup', function (e) {
      if (e.keyCode === 13) {
        $(this).closest('form').submit();
      }
    });
```
This way you get the popup on enter, but as soon as you select an element and hit enter it will also submit the form. Tested with the latest Chrome, Firefox and edge IE. Will test it later on Mac as well.

**Comment by lsimeonov** [comment]
In case this needs to be done with tagging support without triggering the submit on the tag selection. The following should work. **Note the events!** We detach the event on select as it triggers after the keypress. And we attach the event when the select is opened.
Not thoroughly tested!

```javascript
var $input = $('.select2-input');
var $form = $('some-form');

var select2EnterEvent = function (e) {
    if (e.keyCode === 13) {
       $form.submit();
    }
};

$input.on('select2:select', function (e) {
    $form.off('keyup', '.select2-selection', select2EnterEvent);
});

$input.on('select2:opening', function (e) {
    $form.on('keyup', '.select2-selection', select2EnterEvent);
});
```

**Comment by jgdev** [comment]
https://gist.github.com/jgdev/00e1e696231d40741ee1cefb549d5ad2

**Comment by senid231** [comment]
I've managed to keep current behaviour with open & select on enter and made `enter`  to submit only when select2 with closed dropdown is in focus.

works fine with select2 4.0.13

here's sample

```js
$.each( $('.select2'), function () {
  item = $(this);
  item.select2();
  var select2 = item.data('select2');
  var origKeypressCbs = select2.listeners.keypress
  var keypressCb = function (evt) {
    if (evt.key === 'Enter' && !select2.isOpen()) {
      item.closest('form').trigger('submit');
      return
    }

    origKeypressCbs.forEach(function (cb) {
      cb(evt);
    });
  }
  select2.listeners.keypress = [keypressCb];
});
```
